### PR TITLE
[DC-2735] Create cleaning rule to keep `survey_conduct` and `observation` in sync

### PR DESCRIPTION
**Pull request by hiro-mishima**

### CODE REVIEW

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_survey_conduct_ids_test.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        tables_and_counts = [{
+            'fq_table_name':
+                f'{self.project_id}.{self.dataset_id}.{SURVEY_CONDUCT}',
+            'fq_sandbox_table_name':
+                self.fq_sandbox_table_names[0],
+            'loaded_ids': [1, 2, 3],
+            'sandboxed_ids': [3],
+            'fields': ['survey_conduct_id'],
+            'cleaned_values': [(1,), (2,)]
+        }]
```
Comment (lrwb-aou) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_survey_conduct_ids_test.py:88]: I added the following to verify that the rule is not sandboxing and deleting from the `observation` table as well.  It is more of a future change notice.  What do you think?
[2735_test.txt](https://github.com/all-of-us/curation/files/9686737/2735_test.txt)

File: data_steward/cdr_cleaner/clean_cdr.py [hunk 2]
```diff
@@ -331,6 +335,7 @@
     (StringFieldsSuppression,),
     (AggregateZipCodes,),
     (DeidentifyAIANZip3Values,),
+    (DropOrphanedSurveyConductIds,),
     (DropOrphanedPIDS,),
     (RemoveExtraTables,),  # Should be last cleaning rule to be run
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/clean_cdr.py:340]: I know the ticket didn't ask you to, but I am going to update it.  I think we should also run this rule as part of the `REGISTERED_TIER_DEID_CLEAN_CLEANING_CLASSES` and `CONTROLLED_TIER_DEID_CLEAN_CLEANING_CLASSES` lists after `DropZeroConceptIDs` and before `CleanMappingExtTables`.

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
Looks good, I left a few small suggestions.

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_survey_conduct_ids_test.py (hunk 1 above)
Comment (hiro-mishima) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_survey_conduct_ids_test.py:96]: It's a great idea to ensure nothing changes in `observation`. Let's do it. Thanks for the suggestion!

File: data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_survey_conduct_ids.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+Original Issues: DC-2735
+"""
+
+# Python imports
+import logging
+
+# Project imports
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.cdr_cleaner.clean_cdr import COMBINED, CONTROLLED_TIER_DEID, CONTROLLED_TIER_DEID_CLEAN, QUERY, RDR, REGISTERED_TIER_DEID, REGISTERED_TIER_DEID_CLEAN
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_survey_conduct_ids.py:19]: If you package the import like a tuple, the line will break easily and likely follow the yapf coding style.
```suggestion
from constants.cdr_cleaner.clean_cdr import (COMBINED, CONTROLLED_TIER_DEID, CONTROLLED_TIER_DEID_CLEAN, QUERY, RDR, REGISTERED_TIER_DEID, REGISTERED_TIER_DEID_CLEAN)
```

**Review by lrwb-aou — APPROVED** [review]
I left a small format suggestion, but nothing major.  It looks good!  Thank you for getting it done so quickly!

**Pull request merged**